src/realm/sync/client.cpp

```diff
@@ -1102,6 +1106,28 @@ bool SessionImpl::is_steady_state_download_message(DownloadBatchState batch_stat
     return false;
 }
 
+void SessionImpl::update_download_estimate(double download_estimate)
+{
+    if (m_state == State::Active) {
+        m_wrapper.m_download_estimate = download_estimate;
+    }
+}
+
+void SessionImpl::notify_download_progress(const std::optional<uint64_t>& transient_changesets_size)
+{
+    if (m_state == State::Active) {
+        if (transient_changesets_size) {
+            m_wrapper.m_transient_downloaded_bytes =
+                m_wrapper.m_transient_downloaded_bytes.value_or(0) + transient_changesets_size.value();
```

Comment:
`value()` seems to be unavailable on iOS 11, which is the minimum version supported by the Swift SDK. It's a similar issue for lines 1844 and 1846.

Reply:
yeah. just dereference it

Reply:
done